### ISSUE TITLE
"An invalid argument was supplied" when calling GetSystemWebProxy()

### ISSUE THREAD

**Issue by dlepp** [phw198/OutlookGoogleCalendarSync#1034]
> Checklist:-
> * [ ] [Upgrade](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Upgrading) to the latest release (alpha if available) ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg)
> * [ ] Reproduce the issue
> * [ ] Complete the information below
> * [ ] Provide your `OGcalsync.log` file (see [how-to](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#locating-your-ogcalsynclog-file))

**OGCS Version**: v2.8.3 
**Edition**: Portable  
**Sync Direction**: O<->G

**Bug Description**  
On start, presented with error message: Application unexpectedly terminated! An invalid argument was supplied
Application closes

**Steps to Reproduce the Issue**  
1. Start portable application

**Any other information**  
This issue arose ~2 mo. ago on version 2.7.0 with no apparent cause.  Updated to 2.8.3 and it persisted.
:paperclip: Attach your `OGcalsync.log` file.
[OGcalsync2.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/4870738/OGcalsync2.log)

**Comment by phw198** [comment]
Looks like there's a problem when it tries to get the system proxy configuration with `GetSystemWebProxy()` which is a built in system function. If you remove the `settings.xml` does OGCS load, but as soon as your set "Inherit proxy settings from IE" it stops working?

```
2020-07-03 12:51:52,195 INFO  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Using default proxy (app.config / IE).
2020-07-03 12:51:52,195 INFO  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Setting system-wide proxy.
2020-07-03 12:51:52,203 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.Net.Sockets.SocketException: An invalid argument was supplied
2020-07-03 12:51:52,203 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x00002726;10022
2020-07-03 12:51:52,203 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -     at System.Net.SafeCloseSocketAndEvent.CreateWSASocketWithEvent(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType, Boolean autoReset, Boolean signaled)
   at System.Net.NetworkAddressChangePolled..ctor()
   at System.Net.AutoWebProxyScriptEngine.AutoDetector.Initialize()
   at System.Net.AutoWebProxyScriptEngine.AutoDetector.get_CurrentAutoDetector()
   at System.Net.AutoWebProxyScriptEngine..ctor(WebProxy proxy, Boolean useRegistry)
   at System.Net.WebProxy.UnsafeUpdateFromRegistry()
   at System.Net.WebRequest.InternalGetSystemWebProxy()
   at System.Net.WebRequest.GetSystemWebProxy()
   at OutlookGoogleCalendarSync.SettingsProxy.Configure()
   at OutlookGoogleCalendarSync.Program.Main(String[] args)
2020-07-03 12:51:52,213 FATAL  1 OutlookGoogleCalendarSync.Program [0] -  Application unexpectedly terminated!
```

**Comment by dlepp** [comment]
I apologize for the late reply.  If I delete the settings.xml, it still doesn't load.  A new settings file is generated and the same error is produced, shortly after the splash screen. I've attached the new log, which looks like the same errors.

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/4914573/OGcalsync.log)

**Comment by dlepp** [comment]
Just wondering if you have any further suggestions how to resolve this issue?  Thanks!

**Comment by phw198** [comment]
Could you try editing the `settings.xml` file manually and changing the Proxy `Type` from `IE` to `None`:
```
  <Proxy>
...
    <Type>None</Type>
```
Then start up OGCS. It may be that you need to inherit the settings from IE, but this should at least get OGCS started. You could then try setting your proxy settings manually within OGCS too...

**Comment by dlepp** [comment]
Thanks that did get the program started.  If I go to the proxy settings and select "Inherit from IE", it throws an error again.   If I leave it set as "none", and try to sync, I receive the message:

Unable to communicate with Google services. The underlying connection was closed: Unable to connect to the remote server.
Google handshake failed.
Sync encountered a problem and did not complete successfully.

I have attached an updated log file.

[OGcalsync - Copy.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/5054602/OGcalsync.-.Copy.log)
